### DISCUSSION TITLE
How do you set smooth scrolling?

### DISCUSSION THREAD

**Discussion by AustinGitHub** [ajaxorg/ace#4877] [Q&A] [answered]
Hi, 
I am trying to use ace editor in reactJS and I can't seem to figure out the smooth scrolling. I tried turning blockScrolling off but this didn't appear to work. I am unsure if there are any other parameters I can try or if there is a bug for this? 
I can't share much of my code as I can't publicize it, but I have these as my editor options 
```
        enableSnippets: props.isSnippetsEnabled,
        showLineNumbers: false,
        tabSize: 2,
        showPrintMargin: false,
        useWorker: true,
        readOnly: true,
```
```
 editorProps={{ $blockScrolling: false }}
 ```
 
and like I said, I tried the $blockScrolling but this didn't work. But I think this might also be outdated because I don't see it referenced in this repos code (might have been on an old repo I saw this on from my search results of trying to fix this). I will try on a sandbox some point this week when I am not busy, but if anyone knows in the meantime I'd appreciate the help, thanks!

I set up a sandbox here - https://codesandbox.io/s/condescending-water-s0wws7?file=/src/App.js let me know if there are issues with the link. I see the scrolling is still in blocks

**Comment by nightwing** [comment, marked as answer]
` $blockScrolling` is an internal private variable that should not be changed.
There is an `animatedScroll` option `editor.getOption("animatedScroll", true)` which controls the behavior of scrolling during large jumps like `ctrl-home`/`ctrl-end`.

But what exactly do you mean by "smooth scrolling"?

**Reply by AustinGitHub** [reply]
I will try this. For smooth scrolling, I mean like normal scrolling like on a web page rather than block scrolling which I am seeing inside of the ace editor
